open-metadata-implementation/adapters/governance-engines-plugins/gaian-ranger-plugin/src/main/java/org/odpi/openmetadata/adapters/governanceenginesplugins/gaianrangerplugin/RangerPolicyResultFilter.java

```diff
@@ -127,155 +282,142 @@ public boolean setQueriedColumns(int[] queriedColumns) {
+        if (userDetails != null) {
+            logger.logInfo(userDetails.toString());
+            return getUserGroupsByUserId(rangerURL, userDetails);
+        }
+
+        return null;
     }
 
+    private Set<String> getUserGroupsByUserId(String rangerURL, RangerUser userDetails) {
+        String userGroupsUrl = getUserGroupURL(rangerURL);
+        final RangerUser userGroups = getRangerUser(userGroupsUrl, userDetails.getId());
 
-    /****************************************************************************************************************************************************************
-     * 									DEPRECATED / UNUSED METHODS - REQUIRED ONLY FOR COMPATIBILITY WITH 'SQLResultFilter'
-     ****************************************************************************************************************************************************************/
+        if (userGroups != null) {
+            logger.logInfo(userGroups.toString());
+            return userGroups.getGroupNameList();
+        }
 
-    /**
-     * This method is deprecated in favour of the same method below having 3 arguments - it is here for compatibility with SQLResultFilter
-     */
-    public int nextQueriedDataSource(String dataSource, int[] columnMappings) {
-        logger.logDetail("Entered nextQueriedDataSource() (unexpectedly), dataSource: " + dataSource + ", columnMappings: " + Util.intArrayAsString(columnMappings));
-        return -1; // allow all records to be returned (i.e. don't impose a maximum number)
+        return null;
     }
 
-    /**
-     * This method is not currently called by Gaian.
-     */
-    public int setDataSourceWrapper(String wrapperID) {
-        logger.logDetail("Entered setDataSourceWrapper() (unexpectedly), wrapperID: " + wrapperID);
-        return -1; // allow a maximum number of records to be returned
+    private String getUserGroupURL(String rangerURL) {
+        return rangerURL + "/service/xusers/secure/users/{0}";
     }
 
-    /**
-     * This method is deprecated in favour of filterRowsBatch() - it is here for compatibility with SQLResultFilter
-     */
-    public boolean filterRow(DataValueDescriptor[] row) {
-        logger.logDetail("Entered filterRow() (unexpectedly), row: " + Arrays.asList(row));
-        return true; // allow this record to be returned
+    private String getUserDetailsURL(String rangerURL) {
+        return rangerURL + "/service/xusers/users/userName/{0}";
+    }
+
+    private String getRangerURL() {
+        Properties properties = new Properties();
+
+        try {
+            InputStream resource = RangerPolicyResultFilter.class.getClassLoader().getResourceAsStream("application.properties");
+            if (resource != null) {
+                properties.load(resource);
+                logger.logDetail(properties.toString());
+                final String rangerURL = properties.getProperty("ranger.plugin.gaian.policy.rest.url");
```

Comment:
see prior comment about using the plugin configuration to do this (via servicedef)

Reply:
this part was removed.